subsys/nfc/tnep/ch/common.c

```diff
@@ -0,0 +1,418 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include <nfc/ndef/ch_msg.h>
+#include <nfc/ndef/ch_rec_parser.h>
+#include <nfc/ndef/msg_parser.h>
+
+#include <nfc/tnep/tag.h>
+
+#include <logging/log.h>
+
+#include "common.h"
+
+LOG_MODULE_REGISTER(nfc_tnep_ch, CONFIG_NFC_TNEP_CH);
+
+#define HR_RANDOM_NUMBER 0x2684
+
+#if defined(CONFIG_NFC_TNEP_TAG)
+#define NFC_CH_MSG_DEF(_name, _max_record_cnt) \
+	NFC_TNEP_TAG_APP_MSG_DEF(_name, _max_record_cnt)
+#else
+#define NFC_CH_MSG_DEF(_name, _max_record_cnt) \
+	NFC_NDEF_MSG_DEF(_name, _max_record_cnt)
+#endif /* defined(CONFIG_NFC_TNEP_TAG) */
+
+#if defined(CONFIG_NFC_TNEP_POLLER)
+#define NFC_TNEP_CH_NON_CARRIER_REC_CNT 2
+#else
+#define NFC_TNEP_CH_NON_CARRIER_REC_CNT 1
+#endif /* defined(CONFIG_NFC_TNEP_POLLER) */
+
+const uint8_t nfc_tnep_ch_svc_uri[] = {
```

Comment:
can it be static?

Reply:
It cannot. It is exposed as extern in `ch.h`